File: cpp_solutions/202. Happy Number.cc
class Solution {
public:
    
    int new_num(int n) {
        string num = to_string(n);
        int ans = 0;
        for(auto it : num) {
            int k = stoi(string(1, it));
            ans+= k * k;
        }
        return ans;
    }
    bool isHappy(int n) {
        
        int its = 0;
        while (its < 30 && n!=1) {
            int k = new_num(n);
            n = k;
            ++its;
        }
        if (n == 1) {return true;}
        return false;
    }
};

File: python3_solution_set0/202. Happy Number.py
##ss
class Solution:
    def isHappy(self, n: int) -> bool:
        
        count = 0
        while n!=1 and count <=10:
            n = sum([int(x)**2 for x in str(n)])
            count+=1
            
        
        if n == 1:
            return True
        else:
            return False
            
